nifi-commons/nifi-sensitive-property-provider/src/main/java/org/apache/nifi/properties/AWSSensitivePropertyProvider.java

```diff
@@ -0,0 +1,377 @@
+     * If credentials/configuration in bootstrap-aws.conf is not fully configured,
+     * attempt to initialize credentials using default AWS credentials/configuration chain.
+     * Note: This does not verify if credentials are valid.
+     */
+    private final void initializeClient() {
+        if (awsBootstrapProperties == null) {
+            logger.warn("Cannot initialize client if awsBootstrapProperties is null");
+            return;
+        }
+        final String accessKeyId = awsBootstrapProperties.getProperty(ACCESS_KEY_PROPS_NAME);
+        final String secretKeyId = awsBootstrapProperties.getProperty(SECRET_KEY_PROPS_NAME);
+        final String region = awsBootstrapProperties.getProperty(REGION_KEY_PROPS_NAME);
+
+        if (StringUtils.isNotBlank(accessKeyId) && StringUtils.isNotBlank(secretKeyId) && StringUtils.isNotBlank(region)) {
+            logger.debug("Credentials/Configuration provided in bootstrap-aws.conf");
+            try {
+                final AwsBasicCredentials credentials = AwsBasicCredentials.create(accessKeyId, secretKeyId);
+                client = KmsClient.builder()
+                        .region(Region.of(region))
+                        .credentialsProvider(StaticCredentialsProvider.create(credentials))
+                        .build();
+            } catch (final KmsException | NullPointerException | IllegalArgumentException e) {
+                final String msg = "Valid configuration/credentials are required to initialize KMS client";
+                logger.error(msg);
+                throw new SensitivePropertyProtectionException(msg, e);
+            }
+        } else {
+            // attempts to initialize client with credentials provider chain
+            logger.debug("Credentials/Configuration not provided in bootstrap-aws.conf, attempting to use default configuration");
+            try {
+                final DefaultCredentialsProvider credentialsProvider = DefaultCredentialsProvider.builder()
+                        .build();
+                // the following is needed to check the default credential builder, if it fails, throws SdkClientException
+                credentialsProvider.resolveCredentials();
+                client = KmsClient.builder()
+                        .credentialsProvider(credentialsProvider)
+                        .build();
+            } catch (final SdkClientException e) {
+                // this exception occurs if default credentials are not provided
+                final String msg = "Valid configuration/credentials are required to initialize KMS client";
+                logger.error(msg);
+                throw new SensitivePropertyProtectionException(msg, e);
+            }
+        }
+    }
+
+    /**
+     * Checks if the client used to communicate with AWS KMS service is open.
+     * @return true if the client has been initialized and open, false otherwise.
+     */
+    private final boolean isClientOpen() {
```

Comment:
Recommend adjusting the naming to `isClientConfigured()` or `isClientInitialized()` to reflect the check more accurately. On the other hand, since it is just a null check, it may be better to just remove the method and use an inline check where needed.

Reply:
I will change the naming, I think this function makes it neater and the use might be expanded upon.